### PR TITLE
New package: widelands-18

### PR DESCRIPTION
**Pull request by pullmoll**
Forgot this addition to common/shlibs in the original pull request.

### CODE REVIEW

**Comment by pullmoll** [comment]
Argh.. As expected now travis-ci timeouts because it hasn't seen output for more than 10 minutes.
Next try will be without redirecting stderr so that some warnings are printed :)

**Comment by pullmoll** [comment]
Hmm.. it looks like now `common/travis/show_files.sh $BOOTSTRAP $ARCH` makes the output exceed the 4 megabytes limit and I can't do anything about it.

File: srcpkgs/widelands/files/widelands [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+# Wrapper script for widelands
+WL_DATA=/usr/share/games/widelans
```
Comment (xtraeme) [inline srcpkgs/widelands/files/widelands]: widelans -> widelands

File: srcpkgs/widelands/template [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	( ${make_cmd} ${makejobs} ${make_build_args} ${make_build_target} ) > $_logfile
+}
+do_install() {
+	local _logfile
+	: ${make_cmd:=make}
+	: ${make_install_target:=install}
+	_logfile=/tmp/${pkgname}-${version}_${revision}-install.log
+	cd build
+	echo "- install output redirected to $_logfile"
+	( ${make_cmd} DESTDIR=${DESTDIR} ${make_install_args} ${make_install_target} ) > $_logfile
```
Comment (xtraeme) [inline srcpkgs/widelands/template]: This `do_build` and `do_install` stuff is totally crap, remove this completely.

File: srcpkgs/widelands/template [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+post_install() {
+	vlicense COPYING
```
Comment (xtraeme) [inline srcpkgs/widelands/template]: Again, if license is GPL there's no need to install license. The GPL licenses are already part of `base-files`.

File: srcpkgs/widelands/template [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	vinstall ${FILESDIR}/${pkgname}.desktop 644 usr/share/applications
+	vbin ${FILESDIR}/${pkgname}
+}
+do_clean() {
+	# FIXME: archive these logs somewhere?
+	rm -f -v /tmp/${pkgname}-${version}_${revision}-{build,install}.log
```
Comment (xtraeme) [inline srcpkgs/widelands/template]: Ditto for `do_clean`.

xbps-src already creates logs for any phase in ${XBPS_STATEDIR}, aka `${XBPS_BUILDDIR}/.xbps-<pkgname>`.

A package should not do this manually.

File: srcpkgs/widelands/template [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Template file for 'widelands'
+pkgname=widelands
+version=18
+revision=1
+wrksrc=${pkgname}-build${version}-src
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=Release -DLUA_LIBRARY=/usr/lib/liblua5.1.a -DLUA_INCLUDE_DIR=/usr/include/lua5.1"
```
Comment (xtraeme) [inline srcpkgs/widelands/template]: - CMAKE_BUILD_TYPE=Release is already set in `common/build-style/cmake.sh`.

File: srcpkgs/widelands/template [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+hostmakedepends="cmake pkg-config"
+makedepends="boost-devel doxygen python minizip-devel libGL glew-devel ncurses-devel readline-devel SDL-devel SDL_gfx-devel SDL_image-devel SDL_net-devel SDL_mixer-devel SDL_ttf-devel lua51-devel libpng-devel gettext zlib-devel"
```
Comment (xtraeme) [inline srcpkgs/widelands/template]: - I really doubt doxygen is required to build.
- python must be in hostmakedepends instead.
- remove libGL and gettext

File: srcpkgs/widelands/template (hunk 4 above)
Comment (pullmoll) [inline srcpkgs/widelands/template]: I did this just to make travis-ci not terminate the build due to verbosity. This build always hits the 4 megabytes limit.

Comment (xtraeme) [inline srcpkgs/widelands/template]: travis-ci is not ready to handle such big packages. The source packages must not contain workarounds for travis.

File: common/hooks/post-install/03-remove-empty-dirs.sh [hunk 7]
```diff
@@ -1,12 +1,14 @@
 # This hooks removes empty dirs and warns about them.
 
 hook() {
-	for f in $(find ${PKGDESTDIR} -type d -empty|sort -r); do
+	for f in $(find ${PKGDESTDIR} -type d -empty|sort -r|tr " " "|"); do
+		f=${f//|/ }
```
Comment (xtraeme) [inline common/hooks/post-install/03-remove-empty-dirs.sh]: I'd want this var quoted, just in case.

File: srcpkgs/widelands/template (hunk 6 above)
Comment (xtraeme) [inline srcpkgs/widelands/template]: There's a lot of redundant dependencies in this line:

- readline-devel depends on ncurses-devel
- SDL_xxx-devel depends on SDL-devel

File: srcpkgs/widelands/template (hunk 4 above)
Comment (pullmoll) [inline srcpkgs/widelands/template]: Is there a way to tell travis-ci to not even try to build a package?

Comment (xtraeme) [inline srcpkgs/widelands/template]: There's no way AFAIK.

File: srcpkgs/widelands/template (hunk 6 above)
Comment (pullmoll) [inline srcpkgs/widelands/template]: True. And I think png-devel depends on zlib-devel. I'll remove these and build locally before pushing the changes.

File: common/hooks/post-install/03-remove-empty-dirs.sh (hunk 7 above)
Comment (xtraeme) [inline common/hooks/post-install/03-remove-empty-dirs.sh]: Please make this one a separate pull request, I'd like to apply this independently.

File: srcpkgs/widelands/template (hunk 6 above)
Comment (pullmoll) [inline srcpkgs/widelands/template]: Ditto for SDL_mixer-devel. I guess I need to keep the SDL_xxx-devel in makedepends.

Comment (pullmoll) [inline srcpkgs/widelands/template]: DUH! *facepalms* Of course the other way round: remove SDL-devel. I should get a coffee first :)

File: common/hooks/post-install/03-remove-empty-dirs.sh (hunk 7 above)
Comment (pullmoll) [inline common/hooks/post-install/03-remove-empty-dirs.sh]: Done. I'll also create a clean new pull request for widelands-18 after this.

**Comment by chneukirchen** [comment]
FTR, you can use "xmindep pkg1 pkg2 pkg3" from xtools to get the minimal set of dependencies.

**Comment by pullmoll** [comment]
@chneukirchen Great stuff. So using `xmindep $(grep makedepends= srcpkgs/mypkg/template|cut -d'"' -f2)` besides xlint is on my record.
